toothpick-runtime/src/main/java/toothpick/InternalProviderImpl.java

```diff
@@ -89,7 +90,8 @@ public synchronized T get(Scope scope) {
         return instance;
       }
       if (providerFactory.hasScopeAnnotation()) {
-        providerInstance = providerFactory.createInstance(scope);
+        Scope targetScope = providerFactory.getTargetScope(scope);
+        providerInstance = providerFactory.createInstance(targetScope);
```

Comment:
I don't understand these changes. The first line of the factory will do exactly that... The scope param of createInstance is immediatly assigned to the target scope of the factory. How can it solve anything ?